### ISSUE TITLE
Fix S6535 (`unnecessary-character-escapes`): Disambiguate conflicting message ids

### ISSUE THREAD

**Issue by ilia-kebets-sonarsource** [SonarSource/SonarJS#3987]
When using a [`Regexp()` constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/RegExp) and providing a string as parameter, you need to escape special characters like character classes with 2 backslashes `\\`.
Using strings is necessary when you want to make the regex pattern dynamic like:

```js
new RegExp(`\d ${foo}`);
```

It can also happen, that the user started writing his regex in a literal, and copied it to a constructor when making it dynamic. In this case, the rule will flag the backslash with the message `Unnecessary escape character: \d.`
This can be confusing if you were indeed wishing to escape it correctly. There a quickfix in the `no-useless-escape` ESLint implementation that we use that proposes to add the missing backslash, but the error message could be improved to hint to it.

We could also make it clearer in the RSPEC.

The quickfix message in SL for VS code seems to fail to interpolate the variable name:

<img width="917" alt="Screenshot 2023-06-27 at 10 55 05" src="https://github.com/SonarSource/SonarJS/assets/104737176/a990e979-9d3f-4377-aa15-7a74fa44bf8a">

### Special cases

We should take into account cases like that:

```js
const pattern="\d";
const regex = new Regexp(pattern);
```
